### PR TITLE
docs: improve UDF creation documentation

### PR DESCRIPTION
**Pull request by gaurav274**
Updated the docs for creating UDF in eva

### CODE REVIEW

File: docs/source/reference/udfs/custom.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+An abstract method that must be implemented in your child class. The `forward` function receives the frames and runs the Deep Learning model on the frames. The logic for transforming the frames and running the models must be provided by you. The arguments that need to be passed are:
+
+- `input_signatures`: List[IOColumnArgument]
+
+  Data types of the inputs to the `forward` function must be specified. If no constraints are given, then no validation is done for the inputs.
+
+- `output_signatures`: List[IOColumnArgument]
+
+  Data types of the outputs to the `forward` function must be specified. If no constraints are given, then no validation is done for the inputs.
```
Comment (IshSiva) [inline docs/source/reference/udfs/custom.rst:67]: Data types of the outputs _from_ the forward function must be specified.  If no constraints are given, no validation is done for the inputs.

File: docs/source/reference/udfs/custom.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        self.model.eval()
+
+Forward
+--------
+
+An abstract method that must be implemented in your child class. The `forward` function receives the frames and runs the Deep Learning model on the frames. The logic for transforming the frames and running the models must be provided by you. The arguments that need to be passed are:
+
+- `input_signatures`: List[IOColumnArgument]
+
+  Data types of the inputs to the `forward` function must be specified. If no constraints are given, then no validation is done for the inputs.
```
Comment (IshSiva) [inline docs/source/reference/udfs/custom.rst:63]: If no constraints are given, no validation is done for the inputs.

File: docs/source/reference/udfs/custom.rst [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+.. code-block:: python
+
+    @setup(cachable=True, udf_type="object_detection", batchable=True)
+    def setup(self, threshold=0.85):
+        self.threshold = threshold
+        self.model = torchvision.models.detection.fasterrcnn_resnet50_fpn(
+            weights="COCO_V1", progress=False
+        )
+        self.model.eval()
+
```
Comment (IshSiva) [inline docs/source/reference/udfs/custom.rst:55]: Can we add a description of the created UDF explaining the parameters that are set? For example, here we can add,
In this example, we create a UDF for object detection and set the cacheable and batchable parameters to True. `self.threshold`, `self.model` and `self.model.eval()` are the custom logic for this UDF.

File: docs/source/reference/udfs/custom.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        for prediction in predictions:
+            pred_class = [
+                str(self.labels[i]) for i in list(self.as_numpy(prediction["labels"]))
+            ]
+            pred_boxes = [
+                [i[0], i[1], i[2], i[3]]
+                for i in list(self.as_numpy(prediction["boxes"]))
+            ]
+
+
```
Comment (IshSiva) [inline docs/source/reference/udfs/custom.rst:106]: can we give an explanation of the example like the one for setup? 
The input to the forward function is a PyTorch tensor of shape (1, 3, 540, 960) and type Float32. The output is a pandas dataframe that has 3 columns with names "labels", "bboxes" and "scores" and type string, float32 and float32 respectively.

**Comment by IshSiva** [comment]
The documentation looks great!
